src/main/java/com/sequenceiq/cloudbreak/converter/TemplateGroupConverter.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.cloudbreak.converter;
+
+import java.util.Collection;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.controller.BadRequestException;
+import com.sequenceiq.cloudbreak.controller.json.InstanceGroupJson;
+import com.sequenceiq.cloudbreak.domain.Stack;
+import com.sequenceiq.cloudbreak.domain.InstanceGroup;
+import com.sequenceiq.cloudbreak.repository.TemplateRepository;
+
+@Component
+public class TemplateGroupConverter extends AbstractConverter<InstanceGroupJson, InstanceGroup> {
+    @Autowired
+    private TemplateRepository templateRepository;
+
+    @Override
+    public InstanceGroupJson convert(InstanceGroup entity) {
+        InstanceGroupJson instanceGroupJson = new InstanceGroupJson();
+        instanceGroupJson.setGroup(entity.getGroupName());
+        instanceGroupJson.setId(entity.getId());
+        instanceGroupJson.setNodeCount(Integer.parseInt(entity.getNodeCount().toString()));
+        instanceGroupJson.setTemplateId(entity.getTemplate().getId());
+        return instanceGroupJson;
+    }
+
+    @Override
+    public InstanceGroup convert(InstanceGroupJson json) {
+        InstanceGroup instanceGroup = new InstanceGroup();
+        instanceGroup.setGroupName(json.getGroup());
+        instanceGroup.setNodeCount(Integer.valueOf(String.valueOf(json.getNodeCount())));
+        if (!json.getGroup().contains("slave") && json.getNodeCount() != 1) {
```

Comment:
is it a valid assumption that if an instanceGroup's name does not contain slave (like datanodes, workers, etc...) than it has to contain only one node?